staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto

```diff
@@ -1694,6 +1694,8 @@ message WindowsMemoryUsage {
     UInt64Value available_bytes = 3;
     // Cumulative number of page faults.
     UInt64Value page_faults = 4;
+    // Total commit memory in use. Commit memory is total of physical and virtual memory in use.
```

Comment:
I still don't understand why wouldn't it be best to change comment as suggested here: https://github.com/kubernetes/kubernetes/pull/117127#discussion_r1244486342 in this same commit. Changing implementation while comment stays to be incorrect doesn't seem to be a good approach. Sorry if I am missing something

Reply:
We are NOT changing any behavior with adding this field. hcsshim does not even expose or return this commit memory. I feel adding and changing the comment of available memory even before we expose the commit memory used is incorrect. We it would be good to change it only once we actually populate and expose this field from the shim.

Reply:
please refer to https://github.com/kubernetes/kubernetes/pull/117127#discussion_r1228816818
https://github.com/kubernetes/kubernetes/pull/117127#discussion_r1229730904

Reply:
> We are NOT changing any behavior with adding this field. hcsshim does not even expose or return this commit memory. I feel adding and changing the comment of available memory even before we expose the commit memory used is incorrect. We it would be good to change it only once we actually populate and expose this field from the shim.

I would +1 for @SergeyKanzhelev as CRI definition is a guide for implementations.
- If we plan to do that which is designed or in scheduling, we should let others know the CRI design in this spec.

kubelet can implement things based on the design using commit memory before the implementation in container runtimes and other parts. 

> I do think we will want to update available_bytes to be limit - committed_memory once commit memory is plumbed through all the layers. from https://github.com/kubernetes/kubernetes/pull/117127#discussion_r1228816818.

Reply:
There has been quite a bit of back and forth for the past two months on this - I'll update the comment now and update the PR.
However, IMO I would preferred to have updated the code comment only after the values are plumbed through as the behavior till then for available bytes is still going to max - working set memory and might cause more confusion if updated now since there this variable is not even getting populated today.

Reply:
If there are some special changes here， that should be mentioned as well if necessary.
- comments is intented to make things clearer， not into a mess.

Reply:
You can change the comment into the new behavior and add a note after the intended behavior that the semantic of this field was different before.

As @pacoxu said CRI interface express the intended bnehavior and should be changed first